example/Benchmark_kokkos.cpp

```diff
@@ -0,0 +1,208 @@
+  }
+  fs::path base_output_path(argv[1]);
+  // if the folder does not exist, try to make one
+  if (!fs::exists(base_output_path)) {
+    if (!fs::create_directories(base_output_path)) {
+      std::cerr << "Failed to create output directory: " << base_output_path << "\n";
+      return 1;
+    }
+  }
+  int nRho = std::stoi(
+      argv[2]); // test on batch sizes of 144, 269, 512, 2048 or round powers of ten
+  int nT = std::stoi(argv[3]);
+  std::string sp5file = argv[4];
+
+  std::vector<int> matids;
+  for (int i = 5; i < argc; ++i)
+    matids.push_back(std::atoi(argv[i]));
+
+  // == Iterate through each material ==
+  for (int matid : matids) {
+    const int ntimes = 20; // Perform 20 time trials
+
+    // == Get material metadata ==
+    SesameMetadata meta;
+    eosGetMetadata(matid, meta);
+
+    // == Set up bounds ==
+    Real rhoMin = 1.1 * std::max(meta.rhoMin, 1e-5);
+    Real rhoMax = 0.9 * meta.rhoMax;
+    Real TMin = 1.1 * std::max(meta.TMin, 1.0);
+    Real TMax = 0.9 * meta.TMax;
+    Bounds rhoBounds(rhoMin, rhoMax, nRho);
+    Bounds TBounds(TMin, TMax, nT);
+
+    SpinerEOSDependsRhoT eos_rt(sp5file, matid);
+    SpinerEOSDependsRhoSie eos_rs(
+        sp5file,
+        matid); // eventually make it SpinerEOSDependsRhoSie<NullTransfom>, for example
+    EOSPAC eos_ref(matid);
+
+    // These vectors will store the compute time for each model for the 20 trials
+    std::vector<double> time_sie_rt_list;
+
+    Kokkos::View<Real *> rhos_gpu("rhos_d", nRho); // DOES THE FOLLOWING LINES WORK OKAY?
+    Kokkos::View<Real *> temps_gpu("temps_d", nT);
+
+    auto rhos_h = Kokkos::create_mirror_view(rhos_gpu);
+    auto temps_h = Kokkos::create_mirror_view(temps_gpu);
+
+    for (int i = 0; i < nRho; ++i)
+      rhos_h[i] = rhoBounds.i2lin(i);
+    for (int j = 0; j < nT; ++j)
+      temps_h[j] = TBounds.i2lin(j);
+
+    Kokkos::deep_copy(rhos_gpu, rhos_h);
+    Kokkos::deep_copy(temps_gpu, temps_h);
+
+    Kokkos::View<Real **> sie_rt_gpu("sie_rt_d", nT, nRho);
+
+    auto sie_rt_h = Kokkos::create_mirror_view(sie_rt_gpu);
+
+    // move eos to device
+    eos_rt = eos_rt.GetOnDevice();
+
+    auto t0 = high_resolution_clock::now(); // e(rho, T)
+    Kokkos::parallel_for(
+        "sie_rt", Kokkos::MDRangePolicy({0, 0}, {nT, nRho}),
+        KOKKOS_LAMBDA(const int it, const int ir) {
+          sie_rt_gpu(it, ir) =
+              eos_rt.InternalEnergyFromDensityTemperature(rhos_gpu(ir), temps_gpu(it));
+          //          sie_rt_gpu(ir, it) =
+          //              eos_rt.InternalEnergyFromDensityTemperature(rhos_gpu(ir),
+          //              temps_gpu(it));
+        });
+    Kokkos::fence();
+    auto t1 = high_resolution_clock::now();
+    auto elapsed = t1 - t0;
+    time_sie_rt_list.push_back(elapsed.count());
+    // may want to time this seperately
+    Kokkos::deep_copy(sie_rt_h, sie_rt_gpu);
+    // cleanup eos on device
+    eos_rt.Finalize();
+    printf("i am done with %d\n", matid);
+  }
```

Comment:
There doesn't seem to be any check for correctness. I suppose thats fine if correctness is checked elsewhere.